pkg/apis/keycloak/v1alpha1/keycloak_types.go

```diff
@@ -10,9 +10,14 @@ import (
 // KeycloakSpec defines the desired state of Keycloak
 // +k8s:openapi-gen=true
 type KeycloakSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
-	// Add custom validation using kubebuilder tags: https://book-v1.book.kubebuilder.io/beyond_basics/generating_crd.html
+	ExternalDatabaseSecret string              `json:"externalDatabaseSecret,omitempty"`
+	AdminCredentialSecret  string              `json:"adminCredentialSecret,omitempty"`
+	Extensions             []KeycloakExtension `json:"extensions,omitempty"`
+}
+
+type KeycloakExtension struct {
+	Name    string `json:"name"`
```

Comment:
I think there should be either full URL or Maven coordinates here (`G:A:V`).

Reply:
Ok, i'll change the type of `extensions` to a list of strings. We can then probe the value in the operator to figure out if its either a valid URL or Maven Group:Artifact:Version.